hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java

```diff
@@ -4068,8 +4084,31 @@ public void recommissionRegionServer(final ServerName server,
     // Remove the server from decommissioned (draining) server list.
     String parentZnode = getZooKeeper().getZNodePaths().drainingZNode;
     String node = ZNodePaths.joinZNode(parentZnode, server.getServerName());
+    boolean shouldBePermanentlyDecommissioned = false;
+
+    // Get the binary data in the node and check whether the hostname is marked as
+    // permanently decommissioned or not
+    try {
+      byte[] rawData = ZKUtil.getData(getZooKeeper(), node);
+
+      // Check if the data is present for backwards compatibility, some nodes may not have it
```

Comment:
Most nodes won't have this data as the common use-case is to drain by hostname,port,startcode.